pkg/mcs/resource_manager/server/server.go

```diff
@@ -242,12 +241,12 @@ func (s *Server) AddStartCallback(callbacks ...func()) {
 
 // IsServing returns whether the server is the leader, if there is embedded etcd, or the primary otherwise.
 func (s *Server) IsServing() bool {
-	return s.participant.IsLeader()
+	return !s.IsClosed() && s.participant.IsLeader()
 }
 
 // IsClosed checks if the server loop is closed
 func (s *Server) IsClosed() bool {
-	return atomic.LoadInt64(&s.isServing) == 0
+	return s != nil && atomic.LoadInt64(&s.isServing) == 0
```

Comment:
maybe we also  should rename the atomic `isServing`, because it is different but similar with `IsServing()`

Reply:
We can do it in a separate pr to refactor all of them. Some func and variable names look confusing.